tests/Unit/Client/Image/ImageUrlUploadTest.php

```diff
@@ -17,12 +17,25 @@ final class ImageUrlUploadTest extends TestCase
      */
     public function productImageUpload_withURL(): void
     {
-        $imageUrl = 'https://example.com/image.png';
+        $allowedImageType = [
+            'gif'  => 'GIF',
+            'jpeg' => 'JPEG',
+            'jpg'  => 'JPEG',
+            'png'  => 'PNG',
+            'bmp'  => 'BMP',
+        ];
+
+        $imageUrl       = 'https://example.com/image.png';
+        $imageFormat    = $allowedImageType[
+            strtolower(array_values(array_slice(explode('.', $imageUrl), -1))[0])
+        ];
 
         $productImageUpload = new ImageUrlUpload($imageUrl);
 
         self::assertInstanceOf(ImageUrlUpload::class, $productImageUpload);
+        self::assertAttributeEquals($imageFormat, 'imageFormat', $productImageUpload);
         self::assertAttributeEquals($imageUrl, 'imageUrl', $productImageUpload);
-        self::assertSame(['imageUrl' => $imageUrl], json_decode($productImageUpload->getPostBodyData(), true));
+        // self::assertSame(['imageUrl' => $imageUrl], json_decode($productImageUpload->getPostBodyData(), true));
+        self::assertSame(['imageFormat' => $imageFormat, 'imageData' => $imageData], json_decode($productImageUpload->getPostBodyData(), true));
```

Comment:
This should fix this test

```suggestion
        self::assertSame(['imageFormat' => $imageFormat, 'imageUrl' => $imageUrl], json_decode($productImageUpload->getPostBodyData(), true));
```